src/jit/lclvars.cpp

```diff
@@ -5773,6 +5774,30 @@ void Compiler::lvaAlignFrame()
             lvaIncrementFrameSize(sizeof(void*));
         }
     }
+#endif
+
+    if (STACK_ALIGN > REGSIZE_BYTES)
+    {
+        if (lvaDoneFrameLayout != FINAL_FRAME_LAYOUT)
+        {
+            // If we are not doing final layout, we don't know the exact value of compLclFrameSize
+            // and thus do not know how much we will need to add in order to be aligned.
+            // We add the maximum pad that we could ever have (which is 12)
+            lvaIncrementFrameSize(STACK_ALIGN - REGSIZE_BYTES);
+        }
+
+        // The stack must always be 16 byte aligned.
+        int adjustFrameSize = compLclFrameSize;
```

Comment:
only true for unix/x86 case, so either state that, or move the comment within the `#if` that follows.